pkg/personname/common.go

```diff
@@ -0,0 +1,39 @@
+package personname
+
+// renderWithSeps is a generalized function for rendering PN DCM() string with the
+// correct level of null separators.
+func renderWithSeps(sections []string, separator string, nullSepLevel uint) string {
+	// Keep track of whether we have found any non-zero values.
+	nonZeroFound := false
+
+	// We're going to build the string on this variable. Since we are building it
+	// backwards, we can't use strings.Builder.
+	dcmString := ""
+
+	// It's going to be easier to write the correct values if we iterate backwards over
+	// them, since we need look-ahead to know if there are interior null components
+	// from a non-zero value near the end of the sections.
+	for i := len(sections) - 1; i >= 0; i-- {
+		// Get the section.
+		section := sections[i]
+
+		// Add the section to the dcm string
+		dcmString = section + dcmString
```

Comment:
No action needed, just a note for posterity: typically this sort of insert at the beginning implementation is not very efficient, however since `sections` isn't that long (e.g. len = 3) it's totally ok. If for some reason this fn takes much larger sizes of sections in the future, we can consider a more verbose but more efficient way of doing htis.